### DISCUSSION TITLE
Does switchNetwork actually return a promise?

### DISCUSSION THREAD

**Discussion by lsbyerley** [wevm/wagmi#125] [Q&A]
gm... I have an async function in my app that performs some updates to the ui when the network is changed by the user. A[ccording to the docs](https://wagmi-xyz.vercel.app/docs/hooks/useNetwork#switchnetwork), `switchNework` is supposed to return a promise, but it appears that is not the case. In my function below, the console log statement is logged before the network actually changes. 

```jsx
const onChangeNetwork = async (e, c) => {
  const newNetwork = await switchNetwork(c.id);
  console.log("LOG: newNetwork", newNetwork);
};

// the newNetwork variable is logged immediately and contains an object resembling this: { data: true,  error: undefined}
```

Has anyone else needed to await the switchNetwork function and perform actions after the network updates?

**Comment by tmm** [comment]
Just tested out with inject, WalletConnect, and WalletLink connectors and it works fine.

Also, opened a [PR](https://github.com/tmm/wagmi/pull/127) to change return type of `switchChain` from `boolean` to `Chain`.

**Reply by lsbyerley** [reply]
nice.. i'm only using injected provider for my app. If you have time, do you mind posting how you tested this? I'm not sure why mine isn't awaiting the network change and am curious how you verified this

**Reply by lsbyerley** [reply]
The chainId from the useNetwork hook is still the old network's chainId after awaiting the network change. It seems the networkData from the hook should also have the updated chainId

```jsx
const [ { data: networkData }, switchNetwork ] = useNetwork();

// On network change via the user changing it
const onChangeNetwork = async (c) => {
  const newNetwork = await switchNetwork(c.id);
  console.log("LOG: this is newNetwork", newNetwork?.data?.id);
  console.log("LOG: this is networkData hook", networkData?.chain?.id);
};

```

**Reply by lsbyerley** [reply]
@tmm does this make sense? i can do a workaround in my app, just curious if this is actually an issue or as expected

**Reply by tmm** [reply]
Looks like a promise to me

https://codesandbox.io/s/magical-kirch-qi06o?file=/src/App.tsx

https://user-images.githubusercontent.com/6759464/152716784-25499958-1912-434a-9b01-db5fb08f9af4.mp4
